src/pages/Profiles.vue

```diff
@@ -486,6 +486,40 @@ export default class Profiles extends Vue {
         }
     }
 
+    async fixThunderstoreFile(file: string) {
+        let headerLoc = -1;
+
+        // zip end header starts with a 4 byte signature
+        const zipEndHeader = Buffer.from([0x50, 0x4b, 0x05, 0x06]);
+    
+        // stream until we find the end header
+        await fs.createReadStream(file, 8192, async (stream) => {
+            for await(const data of stream) {
+                const index = data.lastIndexOf(zipEndHeader);
+                if (index !== -1) {
+                    headerLoc += index;
+                    break;
+                } else {
+                    headerLoc += data.length;
+                }
+            }
+        });
+
+        // ensure exitst
+        if (headerLoc === -1) {
+            throw new Error("Failed to find zip end header");
+        }
+
+        // header loc is size, so we must add 1 since it's 0 indexed
+        headerLoc += 1;
+
+        // trim file
+        // zip end header is 22 bytes long
```

Comment:
Is it guaranteed to always be 22 bytes long? Is there any case where it could be 21/23?

Reply:
yes https://en.wikipedia.org/wiki/ZIP_(file_format)#End_of_central_directory_record_(EOCD) (note that for some reason wikipedia's using the other endianness for each thing, the bytes for the start of the header are in the right order), 

unless it's a zip64, idk if we need to handle those tho, are they ever created?